### PR TITLE
Bug 1216591 - Support IFNDEF on Gaia preprocessor. r=rchien

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=667f308dc0a38b2d63838145abe43087e5acd0f2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9d2847686f7022f6b6bd99d4f15c61c1e5d6fcb3

File: build/preprocessor.js [hunk 1]
```diff
@@ -18,44 +18,63 @@ function removeFiles(stagePath, list) {
...
   }
 
+  replaced = replaced.replace(regexp, '');
   return replaced;
 }
 
 function processFiles(flag, enable, list, stagePath) {
   list.forEach((fileName) => {
     fileName.unshift(stagePath);
+    utils.log('PREPROCESSOR', flag, enable, fileName);
```
Comment (rickychien) [inline build/preprocessor.js]: I prefer to use lowercase ```preprocessor``` here since all of build logs are lowercase.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2d757479ddb58e4569a1a7510df7d816c26d2c54

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2ed116bc515975184fafdc5829e11ed4c1ff7d8d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c7202a87147f1a564b3f7937fb3c3dc4ebbe8c25

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0c24554a26ad65c26a39b9e1926c1d472b3d4a4f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ce9bad0cbe23ea5f679a29f3a77e7fd169c01a17

**Pull request merged**